packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -203,7 +204,10 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
         telemetryLogger: ITelemetryLogger,
         timeoutMs: number,
         epochTracker: EpochTracker,
-        socketReferenceKeyPrefix: string | undefined): Promise<OdspDocumentDeltaConnection> {
+        socketReferenceKeyPrefix: string | undefined,
+        sessionJoinCache: PromiseCacheWithOneHourSlidingExpiry<ISocketStorageDiscovery>,
```

Comment:
If possible, this should be PromiseCache<string, ISocketStorageDiscovery>, not to encode implementation details here, only interface.
I'd go even further, and abstract away sessionJoinCache & joinSessionCacheKey via

joinCacheRemoveCallback: () =>void,

There is just no need to have that kind of implementation leakage that far. Maybe go even further and provide more general callback:

onDisconnect: (error: OdspError) => error,

And let the caller encode specialness of DriverErrorType.authorizationError

Reply:
I think it would be better for this logic not to be here at all, but rather be in connectToDeltaStream().
Similar how we already subscribe to "op" vent there, we can subscribe to "disconnect" event and test error for right condition to remove join session cache. It would be broader check, but I think it's Ok (I do not see cases where it would cause trouble) and it would be in the right place, without polluting lower levels.